### PR TITLE
Update CONTRIBUTING

### LINKED ISSUE #487: CONTRIBUTING.md out of date
## Checklist

- [ ] I'm reporting a broken site support issue
- [ ] ~I've verified that I'm running yt-dlp version **2021.07.07**~
- [ ] ~I've checked that all provided URLs are alive and playable in a browser~
- [ ] ~I've checked that all URLs and arguments with special characters are properly quoted or escaped~
- [X] I've searched the bugtracker for similar bug reports including closed ones
- [X] I've read bugs section in FAQ

## Description

I want to contribute a feature to yt-dlp so the first thing I did was read CONTRIBUTING.md. I found that it is identical to the youtube_dl CONTRIBUTING.md document (except for one character difference in an example URL). The instructions for running are obviously not correct and from the setup.py it looks like yt-dlp has dropped support for python 2 so other instructions in that document are incorrect.
There are probably other differences in code organisation and code style which differ from youtube_dl so it would be nice to customize this document so that it isn't misleading.

### PR DESCRIPTION
**Pull request by ghost**
---

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [X] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [X] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Related to Issue #487, Initial update of the Contributing Docs

- Find all references to `youtube-dl` and replace with `yt-dlp`.
- Find all references to `youtube_dl` and replace with `yt_dlp`.
- Update Verbose Output example.
- Add Page Header, and Initial Section Header.
- Remove sentence relating to the `--call-home` option
- Update URLs to use `yt-dlp` Project.
- Changed Case of `DEVELOPER INSTRUCTIONS`
- Added comment, to use Issue Templates
- Added comment abut changes in default behavior and options.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -147,45 +153,45 @@ After you have ensured this site is distributing its content legally, you can fo
...
 
  - `id` (media identifier)
  - `title` (media title)
  - `url` (media download URL) or `formats`
 
-In fact only the last option is technically mandatory (i.e. if you can't figure out the download location of the media the extraction does not make any sense). But by convention youtube-dl also treats `id` and `title` as mandatory. Thus the aforementioned metafields are the critical data that the extraction does not make any sense without and if any of them fail to be extracted then the extractor is considered completely broken.
+In fact only the last option is technically mandatory (i.e. if you can't figure out the download location of the media the extraction does not make any sense). But by convention yt-dlp also treats `id` and `title` as mandatory. Thus the aforementioned metafields are the critical data that the extraction does not make any sense without and if any of them fail to be extracted then the extractor is considered completely broken.
```
Comment (pukkandan) [inline CONTRIBUTING.md:192]: url/formats is not mandatory in yt-dlp due to `--ignore-no-formats-error`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -147,45 +153,45 @@ After you have ensured this site is distributing its content legally, you can fo
...
 
-[Any field](https://github.com/ytdl-org/youtube-dl/blob/7f41a598b3fba1bcab2817de64a08941200aa3c8/youtube_dl/extractor/common.py#L188-L303) apart from the aforementioned ones are considered **optional**. That means that extraction should be **tolerant** to situations when sources for these fields can potentially be unavailable (even if they are always available at the moment) and **future-proof** in order not to break the extraction of general purpose mandatory fields.
+[Any field](https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/common.py#L188-L303) apart from the aforementioned ones are considered **optional**. That means that extraction should be **tolerant** to situations when sources for these fields can potentially be unavailable (even if they are always available at the moment) and **future-proof** in order not to break the extraction of general purpose mandatory fields.
```
Comment (pukkandan) [inline CONTRIBUTING.md:194]: The line numbers both here and in any other places may need to be changed

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -407,15 +413,15 @@ Incorrect:
 
 ### Use convenience conversion and parsing functions
 
-Wrap all extracted numeric data into safe functions from [`youtube_dl/utils.py`](https://github.com/ytdl-org/youtube-dl/blob/master/youtube_dl/utils.py): `int_or_none`, `float_or_none`. Use them for string to number conversions as well.
+Wrap all extracted numeric data into safe functions from [`yt_dlp/utils.py`](https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/utils.py): `int_or_none`, `float_or_none`. Use them for string to number conversions as well.
 
 Use `url_or_none` for safe URL processing.
 
 Use `try_get` for safe metadata extraction from parsed JSON.
```
Comment (pukkandan) [inline CONTRIBUTING.md:420]: point to `try_get`, `dict_get` nd `traverse_obj`

**Review by pukkandan — CHANGES_REQUESTED** [review]
Github won't let me comment on unedited lines...
1. Remove `Many contributors, including myself, are also not native speakers, so we may misread some parts.`
1. `As a committer myself, I often get frustrated by these issues` => `we often...`
1. `keep lines of code under 80 characters long` - I think 80chars is a bad limit in current times. 100-120 chars is easily readable on modern screens

---

We also need to decide whether to 
1. add this entirely to the README and use `devscripts/make_contributing.py` (like youtube-dl) OR
2. let it be its own thing and simply link to it in the Readme

---

Note to myself: Improve `More examples`

**Comment by pukkandan** [comment]
Related: https://github.com/ytdl-org/youtube-dl/pull/29519

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -94,29 +103,26 @@ If you want to create a build of youtube-dl yourself, you'll need
 
 ### Adding support for a new site
 
-If you want to add support for a new site, first of all **make sure** this site is **not dedicated to [copyright infringement](README.md#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free)**. youtube-dl does **not support** such sites thus pull requests adding support for them **will be rejected**.
+If you want to add support for a new site, first of all **make sure** this site is **not dedicated to [copyright infringement](README.md#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free)**. yt-dlp does **not support** such sites thus pull requests adding support for them **will be rejected**.
```
Comment (pukkandan) [inline CONTRIBUTING.md:106]: this link doesn't work

**Review by pukkandan — COMMENTED** [review]
https://github.com/ytdl-org/youtube-dl#bugs has:

> Bugs and suggestions should be reported at: https://github.com/ytdl-org/youtube-dl/issues. Unless you were prompted to or there is another pertinent reason (e.g. GitHub fails to accept the bug report), please do not send bug reports via personal email. For discussions, join us in the IRC channel #youtube-dl on freenode (webchat).

We should add a similar message at the top, replacing IRC with the discord link

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -58,23 +65,25 @@ We are then presented with a very complicated request when the original problem
...
 
 ###  Is anyone going to need the feature?
 
 Only post features that you (or an incapacitated friend you can personally talk to) require. Do not post features because they seem like a good idea. If they are really useful, they will be requested by someone who requires them.
 
-###  Is your question about youtube-dl?
+###  Is your question about yt-dlp?
+
+It may sound strange, but some bug reports we receive are completely unrelated to yt-dlp and relate to a different, or even the reporter's own, application. Please make sure that you are actually using yt-dlp. If you are using a UI for yt-dlp, report the bug to the maintainer of the actual application providing the UI. On the other hand, if your UI for yt-dlp fails in some way you believe is related to yt-dlp, by all means, go ahead and report the bug.
```
Comment (Ashish0804) [inline CONTRIBUTING.md:76]:
```suggestion
Some bug reports are completely unrelated to yt-dlp and relate to a different, or even the reporter's own, application. Please make sure that you are actually using yt-dlp. If you are using a UI for yt-dlp, report the bug to the maintainer of the actual application providing the UI. On the other hand, if your UI for yt-dlp fails in some way you believe is related to yt-dlp, by all means, go ahead and report the bug.
```

File: CONTRIBUTING.md (hunk 2 above)
Comment (timethrow) [reply CONTRIBUTING.md]: Updated now

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -30,23 +33,27 @@ So please elaborate on what feature you are requesting, or what bug you want to
...
 
 ###  Is the issue already documented?
 
-Make sure that someone has not already opened the issue you're trying to open. Search at the top of the window or browse the [GitHub Issues](https://github.com/ytdl-org/youtube-dl/search?type=Issues) of this repository. If there is an issue, feel free to write something along the lines of "This affects me as well, with version 2015.01.01. Here is some more information on the issue: ...". While some issues may be old, a new post into them often spurs rapid activity.
+Make sure that someone has not already opened the issue you're trying to open. Search at the top of the window or browse the [GitHub Issues](https://github.com/yt-dlp/yt-dlp/search?type=Issues) of this repository. If there is an issue, feel free to write something along the lines of "This affects me as well, with version 2021.01.01. Here is some more information on the issue: ...". While some issues may be old, a new post into them often spurs rapid activity.
```
[the remark this answers is not in the document]
Comment (timethrow) [reply CONTRIBUTING.md:48]: 5515f08ea15221ef6bacb51571eef3a4b3282109
Done

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -84,7 +93,7 @@ To run the test, simply invoke your favorite test runner, or execute a test file
 
 See item 6 of [new extractor tutorial](#adding-support-for-a-new-site) for how to run extractor specific test cases.
 
-If you want to create a build of youtube-dl yourself, you'll need
+If you want to create a build of yt-dlp yourself, you'll need
```
[the remark this answers is not in the document]
Comment (timethrow) [reply CONTRIBUTING.md]: b73d2285e40d0746f19cef883ee7c65974339a80
Linked to `README.md`

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -94,29 +103,26 @@ If you want to create a build of youtube-dl yourself, you'll need
...
-4. Start with this simple template and save it to `youtube_dl/extractor/yourextractor.py`:
+4. Start with this simple template and save it to `yt_dlp/extractor/yourextractor.py`:
 
     ```python
     # coding: utf-8
-    from __future__ import unicode_literals
-
     from .common import InfoExtractor
 
-
```
[the remark this answers is not in the document]
Comment (timethrow) [reply CONTRIBUTING.md:119]: Readded in e44561ad08c847e803482353285241e55cf49933

**Comment by Ashish0804** [comment]
Please update this line aswell to say `yt-dlp`

[In order to be accepted and merged into youtube-dl each piece ](https://github.com/yt-dlp/yt-dlp/blob/master/.github/PULL_REQUEST_TEMPLATE.md#in-order-to-be-accepted-and-merged-into-youtube-dl-each-piece-of-code-must-be-in-public-domain-or-released-under-unlicense-check-one-of-the-following-options)

Unless the license prohibits it.

**Comment by gamer191** [comment]
Please ignore this comment. It is semi-automated and is intended to ensure PRs by @ghost get indexed by Github

**Pull request merged**